apps/root/src/state/balances/actions.ts

```diff
@@ -1,121 +1,130 @@
   const state = getState() as RootState;
   const storedTokenAddresses = Object.values(state.balances[chainId].balancesAndPrices || {}).map(
     (tokenBalance) => tokenBalance.token.address
   );
   let priceResponse: CurrentPriceForChainResponse = {};
   if (!!storedTokenAddresses.length) {
-    try {
-      dispatch(setLoadingChainPrices({ chainId, isLoading: true }));
-      priceResponse = await sdkService.sdk.priceService.getCurrentPricesForChain({
-        chainId,
-        addresses: storedTokenAddresses,
-      });
-    } catch (e) {
-      console.error(e);
-    }
-    dispatch(setLoadingChainPrices({ chainId, isLoading: false }));
+    priceResponse = await sdkService.sdk.priceService.getCurrentPricesForChain({
+      chainId,
+      addresses: storedTokenAddresses,
+    });
   }
 
   return { chainId, prices: priceResponse };
 });
 
 export const fetchPricesForAllChains = createAsyncThunk<void, void, { extra: ExtraArgument }>(
   'balances/fetchInitialBalances',
-  async (nothing, { getState, dispatch }) => {
+  (_, { getState, dispatch }) => {
     const state = getState() as RootState;
     const chainsWithBalance = Object.keys(state.balances);
 
-    const pricePromises = chainsWithBalance.map(async (chainId) =>
+    const pricePromises = chainsWithBalance.map((chainId) =>
       dispatch(fetchPricesForChain({ chainId: Number(chainId) }))
     );
-    await Promise.all(pricePromises);
+    void Promise.all(pricePromises);
   }
 );
 
-export const fetchInitialBalances = createAsyncThunk<BalancesState, void, { extra: ExtraArgument }>(
-  'balances/fetchInitialBalances',
-  async (nothing, { extra: { web3Service }, getState }) => {
-    const accountService = web3Service.getAccountService();
-    const meanApiService = web3Service.getMeanApiService();
-    const chainIds = Object.values(NETWORKS).map((network) => network.chainId);
-    const wallets = accountService.getWallets().map((wallet) => wallet.address);
-    const state = getState() as RootState;
-    const allTokens = flatten(Object.values(state.tokenLists.byUrl).map((list) => list.tokens));
-    const tokenList: TokenList = keyBy(allTokens, 'address');
+export const fetchInitialBalances = createAsyncThunk<
+  Omit<BalancesState, 'isLoadingAllBalances'>,
+  void,
+  { extra: ExtraArgument }
+>('balances/fetchInitialBalances', async (_, { extra: { web3Service }, getState }) => {
+  const accountService = web3Service.getAccountService();
+  const meanApiService = web3Service.getMeanApiService();
+  const chainIds = Object.values(NETWORKS).map((network) => network.chainId);
+  const wallets = accountService.getWallets().map((wallet) => wallet.address);
+  const state = getState() as RootState;
+  const allTokens = flatten(Object.values(state.tokenLists.byUrl).map((list) => list.tokens));
 
-    const accountBalancesResponse = await meanApiService.getAccountBalances({
-      wallets,
-      chainIds,
-    });
+  const tokenListByChainId = allTokens.reduce(
```

Comment:
Carefull I don't think this will actually take into account the `priority` parameter on the tokenLists